File: app/src/main/java/com/lixm/animationdemo/activity/PayPassportActivity.java
package com.lixm.animationdemo.activity;

import android.os.Bundle;
import android.text.Editable;
import android.text.TextUtils;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.lixm.animationdemo.R;
import com.lixm.animationdemo.customview.CustomPayEditText;
import com.lixm.animationdemo.dialog.PayDialog;

public class PayPassportActivity extends BaseActivity {

    private String TAG = "PayPassportActivity";
    private CustomPayEditText mCustomPayEditText;
    private CustomPayEditText CustomPayEditText2;
    private TextView ellipsizeTxt;
    private EditText ellipsizeEt;
    private Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pay_passport);
        mCustomPayEditText = (CustomPayEditText) findViewById(R.id.ppe_pwd);
        CustomPayEditText2 = (CustomPayEditText) findViewById(R.id.ppe_pwd2);
        button = (Button) findViewById(R.id.btn_dialog);

        mCustomPayEditText.initStyle(R.drawable.edit_num_bg, 6, 0.33f, R.color.color_333, R.color.aaaaa, 20);
        mCustomPayEditText.setOnTextFinishListener(new CustomPayEditText.OnTextFinishListener() {
            @Override
            public void onFinish(String str) {//密码输入完后的回调
                Toast.makeText(PayPassportActivity.this, str, Toast.LENGTH_SHORT).show();
            }
        });

        CustomPayEditText2.initStyle(R.drawable.edit_num_bg_red, 8, 0.33f, R.color.colorAccent, R.color.colorAccent, 20);
        CustomPayEditText2.setOnTextFinishListener(new CustomPayEditText.OnTextFinishListener() {
            @Override
            public void onFinish(String str) {
                Toast.makeText(PayPassportActivity.this, "显示明文：" + str, Toast.LENGTH_SHORT).show();
                CustomPayEditText2.setShowPwd(false);
            }
        });

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                PayDialog payDialog = new PayDialog(PayPassportActivity.this);
                payDialog.show();
            }
        });

        ellipsizeTxt = (TextView) findViewById(R.id.ellipsize_txt);
        ellipsizeEt= (EditText) findViewById(R.id.ellipsize_et);
        ellipsizeEt.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {
                ellipsizeTxt.setText(s.toString());
            }
        });
    }
}

File: app/src/main/java/com/lixm/animationdemo/activity/PayPassportActivity.kt
package com.lixm.animationdemo.activity

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import com.lixm.animationdemo.R
import com.lixm.animationdemo.customview.CustomPayEditText
import com.lixm.animationdemo.dialog.PayDialog

class PayPassportActivity : BaseActivity() {

    private val TAG = "PayPassportActivity"
    private var mCustomPayEditText: CustomPayEditText? = null
    private var CustomPayEditText2: CustomPayEditText? = null
    private var ellipsizeTxt: TextView? = null
    private var ellipsizeEt: EditText? = null
    private var button: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_pay_passport)
        mCustomPayEditText = findViewById<View>(R.id.ppe_pwd) as CustomPayEditText
        CustomPayEditText2 = findViewById<View>(R.id.ppe_pwd2) as CustomPayEditText
        button = findViewById<View>(R.id.btn_dialog) as Button

        mCustomPayEditText!!.initStyle(R.drawable.edit_num_bg, 6, 0.33f, R.color.color_333, R.color.aaaaa, 20)
        mCustomPayEditText!!.setOnTextFinishListener { str -> //密码输入完后的回调
            Toast.makeText(this@PayPassportActivity, str, Toast.LENGTH_SHORT).show()
        }

        CustomPayEditText2!!.initStyle(R.drawable.edit_num_bg_red, 8, 0.33f, R.color.colorAccent, R.color.colorAccent, 20)
        CustomPayEditText2!!.setOnTextFinishListener { str ->
            Toast.makeText(this@PayPassportActivity, "显示明文：" + str, Toast.LENGTH_SHORT).show()
            CustomPayEditText2!!.setShowPwd(false)
        }

        button!!.setOnClickListener {
            val payDialog = PayDialog(this@PayPassportActivity)
            payDialog.show()
        }

        ellipsizeTxt = findViewById<View>(R.id.ellipsize_txt) as TextView
        ellipsizeEt = findViewById<View>(R.id.ellipsize_et) as EditText
        ellipsizeEt!!.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

            }

            override fun afterTextChanged(s: Editable) {
                ellipsizeTxt!!.text = s.toString()
            }
        })
    }
}
